zha/application/platforms/sensor/__init__.py

```diff
@@ -202,14 +202,18 @@ def _init_from_quirks_metadata(self, entity_metadata: ZCLSensorMetadata) -> None
             self._divisor = entity_metadata.divisor
         if entity_metadata.multiplier is not None:
             self._multiplier = entity_metadata.multiplier
+        if entity_metadata.translation_key is not None:
+            self._attr_translation_key = entity_metadata.translation_key
+        if entity_metadata.state_class is not None:
+            self._attr_state_class = entity_metadata.state_class.value
         if entity_metadata.device_class is not None:
             self._attr_device_class = validate_device_class(
                 SensorDeviceClass,
                 entity_metadata.device_class,
                 Platform.SENSOR.value,
                 _LOGGER,
             )
-        if entity_metadata.device_class is None and entity_metadata.unit is not None:
+        if entity_metadata.unit is not None:
```

Comment:
the device class controls the unit when it is set. This is a HA core thing and not controlled by us.

Reply:
I think we still need to provide the unit to let HA know what unit we're using. Take a look at our existing sensors in ZHA:
https://github.com/zigpy/zha/blob/e1ae71cc31cd53700038881de1f4ee51586caca9/zha/application/platforms/sensor/__init__.py#L585-L593

Reply:
This part will be resolved via:
- https://github.com/zigpy/zha/pull/219/